src/index.js

```diff
@@ -553,33 +553,205 @@ export default class ToggleBlock {
+      }
+      counter += 1;
+    });
+    return counter;
+  }
+
+  /**
+   * Highlight the blocks that belongs to the Toggle
+   * @param {string} fk - The id of the root Toggle
+   */
+  highlightToggleItems(fk) {
+    const listChildren = document.querySelectorAll(`div[foreignKey="${fk}"]`);
+    listChildren.forEach((child) => {
+      child.classList.add('ce-block--selected');
+      // Evaluate if the child is a toggle, then highlight also its children
+      if (child.hasAttribute('status')) {
+        const childId = child.querySelector('.toggle-block__selector').getAttribute('id');
+        this.highlightToggleItems(childId);
+      }
+    });
+  }
+
+  /**
+   * Adds events for the move up, move down and delete options in the toolbar
    */
   renderSettings() {
     const settingsBar = document.getElementsByClassName('ce-settings--opened');
     const optionsContainer = settingsBar[0];
     const options = optionsContainer.lastChild;
     const toggleIndex = this.api.blocks.getCurrentBlockIndex();
-    const listChildren = document.querySelectorAll(`div[foreignKey="${this.wrapper.id}"]`);
 
-    for (let i = 0; i < listChildren.length; i += 1) {
-      listChildren[i].classList.add('ce-block--selected');
-    }
+    this.highlightToggleItems(this.wrapper.id);
 
     setTimeout(() => {
+      const moveDownButton = options.getElementsByClassName('ce-tune-move-down')[0];
+      if (moveDownButton) {
+        moveDownButton.addEventListener('click', () => {
+          this.moveToggle(toggleIndex, 0);
+        });
+      }
+
+      const moveUpButton = options.getElementsByClassName('ce-tune-move-up')[0];
+      if (moveUpButton) {
+        moveUpButton.addEventListener('click', () => {
+          this.moveToggle(toggleIndex, 1);
+        });
+      }
+
       const deleteButton = options.getElementsByClassName('ce-settings__button--delete')[0];
-      deleteButton.addEventListener('click', () => {
-        const classesList = deleteButton.classList;
-        const classes = Object.values(classesList);
+      if (deleteButton) {
+        deleteButton.addEventListener('click', () => {
+          const classesList = deleteButton.classList;
+          const classes = Object.values(classesList);
+
+          if (classes.indexOf('clicked-to-destroy-toggle') === -1) {
+            deleteButton.classList.add('clicked-to-destroy-toggle');
+          } else {
+            this.removeFullToggle(toggleIndex);
+          }
+        });
+      }
+    });
+  }
 
-        if (classes.indexOf('clicked-to-destroy-toggle') === -1) {
-          deleteButton.classList.add('clicked-to-destroy-toggle');
+  /**
+   * Move the Toggle with all its children and nested toggles.
+   * Index of the root toggle before it is moved by editorjs core.
+   * @param {number} toggleInitialIndex
+   * @param {number} direction // 0: Move down || 1: Move up
+   */
+  moveToggle(toggleInitialIndex, direction) {
+    if (!this.readOnly) {
+      this.api.toolbar.close();
```

Comment:
In this case, you can use destructuring with the API, for example:

`const { toolbar: { close }, blocks: { getCurrentBlockIndex, getBlocksCount, move } } = this.api;`

Reply:
Apply in all possible cases

Reply:
Done!